Emby.Server.Implementations/Library/UserManager.cs

```diff
@@ -307,11 +311,13 @@ public async Task<User> AuthenticateUser(string username, string password, strin
                     user = Users
                         .FirstOrDefault(i => string.Equals(username, i.Name, StringComparison.OrdinalIgnoreCase));
 
-                    var hasNewUserPolicy = authenticationProvider as IHasNewUserPolicy;
-                    if (hasNewUserPolicy != null)
+                    if (authenticationProvider.GetType() != typeof(InvalidAuthProvider))
```

Comment:
This check seems redundant.

Reply:
It's not - it prevents overwriting the on-disk `policy.xml` with the invalid provider.

Reply:
But `InvalidAuthProvider` doesn't implement `IHasNewUserPolicy`, so it shouldn't ever overwrite, right?

Reply:
@LogicalPhallacy does it?

Reply:
Actually if it doesn't implement IHasNewUserPolicy, wouldn't that throw an exception (possibly a fatal exception) if those code tries to execute? In that case we probably do want to keep this conditional.

Reply:
Nope, it's good without this check. No errors or issues and it doesn't overwrite.